### DISCUSSION TITLE
Is there any possible to implement this library(or framework?) without using js, use something maybe like rust?

### DISCUSSION THREAD

**Discussion by zzz6519003** [bigskysoftware/htmx#2379] [Q&A] [answered]
Is there any possible to implement this library(or framework?) without using js, use something maybe like rust?

**Comment by nickchomey** [comment]
Maybe it could be compiled to wasm, if that's what you're asking. If you're talking about the backend, then you can use any language you want so long as you're sending HTML to the browser.

**Comment by CaliViking** [comment, marked as answer]
@zzz6519003 - if you are referring to the client side (in the web browser), then there is currently no way to achieve this without at least some amount of JavaScript.
The reason is that WebAssembly (which is where you would put the Rust code) cannot manipulate the DOM. This means that there would need to be a JS layer (some JS Glue code) where the DOM is manipulated.
Please see https://developer.mozilla.org/en-US/docs/WebAssembly/Concepts 

You can use anything on the server side. I have been using Rust Axum.

**Reply by zzz6519003** [reply]
i can tell you're a pro of this
